plugins/schedule/base/lib/kScheduleEventsConsumer.php

```diff
@@ -0,0 +1,108 @@
+<?php
+class kScheduleEventsConsumer implements kObjectChangedEventConsumer, kObjectDeletedEventConsumer, kObjectCreatedEventConsumer
+{
+    public function shouldConsumeCreatedEvent(BaseObject $object)
+    {
+        if ($object instanceof categoryEntry)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    public function shouldConsumeDeletedEvent(BaseObject $object)
+    {
+        if ($object instanceof categoryEntry)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    public function shouldConsumeChangedEvent(BaseObject $object, array $modifiedColumns)
+    {
+        try
+        {
+            if ($object instanceof categoryEntry && in_array(categoryEntryPeer::STATUS, $modifiedColumns) && $object->getStatus() == categoryEntryStatus::ACTIVE)
+            {
+                return true;
+            }
+        } catch (Exception $e)
+        {
+            KalturaLog::err('Failed to process shouldConsumeChangedEvent - ' . $e->getMessage());
+        }
+
+        return false;
+    }
+
+    /* (non-PHPdoc)
+     * @see kObjectChangedEventConsumer::objectChanged()
+     */
+    public function objectChanged(BaseObject $object, array $modifiedColumns)
+    {
+        try

```

Comment:
No need to wrap with try and catch because the events manager already does it.
No need to check the object type because the should method already does it.